src/Control/Carrier/Interpret.hs

```diff
@@ -109,6 +109,11 @@ newtype InterpretC s (sig :: (* -> *) -> * -> *) m a =
 instance MonadTrans (InterpretC s sig) where
   lift = InterpretC
 
+instance MonadUnliftIO m => MonadUnliftIO (InterpretC s sig m) where
+  askUnliftIO = InterpretC $ withUnliftIO $ \u -> return (UnliftIO (unliftIO u . runInterpretC))
+  {-# INLINE askUnliftIO #-}
+  withRunInIO inner = InterpretC $ withRunInIO $ \run -> inner (run . runInterpretC)
+  {-# INLINE withRunInIO #-}
```

Comment:
1. This gave me pause for a moment due to the existence of `runInterpretState`—but that relies on an underlying `StateC`, which wouldn’t have a `MonadUnliftIO` instance, so this is orthogonal 👍

2. Just a heads up that I’ve 🔥d `runInterpretC` in #275, so one of us is going to have to address these uses of it.